### DISCUSSION TITLE
p-accordion not fetching correct index of the accordion tab with onOpen event while using nested p-accordion

### DISCUSSION THREAD

**Discussion by dibyendusaha** [primefaces/community#850] [PrimeNG] [answered]
I have created an example to explain the scenario in a better way.
Please find the below link to find the example:
[https://stackblitz.com/edit/stackblitz-starters-jhq5ug](https://stackblitz.com/edit/stackblitz-starters-jhq5ug?file=src%2Fapp%2Fapp.component.ts)
[Alternate link](https://stackblitz.com/edit/stackblitz-starters-jhq5ug?description=An%20angular-cli%20project%20based%20on%20@angular/animations,%20@angular/common,%20@angular/compiler,%20@angular/core,%20@angular/forms,%20@angular/platform-browser,%20@angular/platform-browser-dynamic,%20@angular/router,%20core-js,%20rxjs,%20tslib%20and%20zone.js&file=src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.scss&title=Angular%20Starter)

Angular version used 17.0.8
PrimeNG version used 17.3.1

I am seeing this issue since version 16.X.X

Trying to replicate the scenario though below diagram:

- ⬇️ Accordion 1

-     - Accordion 1.1

        - Accordion 1.1.1
        - Accordion 1.1.2
        - Accordion 1.1.3
        - Accordion 1.1.4

-     - Accordion 1.2

        - Accordion 1.2.1
        - Accordion 1.2.2
        - Accordion 1.2.3

- ⬇️ Accordion 2

-     - Accordion 2.1

        - Accordion 2.1.1
        - Accordion 2.1.2

Expected behaviour is:
When I am opening Accordion 1, then onOpen event should return _event.index_ as 0 - it is returning as expected ✅
When I am opening Accordion 2, then onOpen event should return _event.index_ as 1 - it is not happening ❌, rather it is returning 10

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Nested accordions cause several technical issues due to angular animations. One of them is the issue you've mentioned above. In our roadmap, we have plans to implement custom animations to replace angular animations and fix issues raised by nested animations. Unfortunately, we cannot provide a solution for now due to the technical limitations.
